addon/matcher.js

```diff
@@ -87,11 +87,46 @@ function processMatchedGroups(matchedGroups, addonSettings, eventType, eventPara
       if (eventType === 'transition' && addonSettings.updateDocumentLocationOnTransitions) {
         matchedGroup.tracker.set('location', document.URL);
       }
+
+      processTiming(matchedGroup, eventType, eventValue);
+
       // handle particular (matched) insight
       matchedGroup.handler(eventType, eventParams, matchedGroup.tracker);
     }
 }
 
+function processTiming(matchedGroup, eventType, eventValue) {
+  if (!(window.performance && window.performance.mark)) {
+    Ember.warn("Your browser doesn't support User Timing API!");
+    return;
+  }
+
+  var currentEvent = eventType.substr(0, 1) + ':' + eventValue;
+  window.performance.mark(currentEvent);
+
+  var prevLists;
+  if (matchedGroup.timingsConfig) {
+    prevLists = matchedGroup.timingsConfig[currentEvent];
+  }
+
+  if (prevLists) {
+    prevLists.forEach(function(prevList) {
+      for (var i=0, len=prevList.prevPoints.length; i<len; i++) {
+        var fromEvent = prevList.prevPoints[i];
+        var saved = window.performance.getEntriesByName(fromEvent);
+
+        if (saved.length) {
+          var groupPrefix = matchedGroup.name ? matchedGroup.name + '_' : '';
+          var measureName = groupPrefix + fromEvent + '_' + currentEvent;
```

Comment:
Could be potential issue with actions/transitions with `_` in their names. I guess the much better delimiter could be `..` or `~`/`<>` or smth like that.